### PR TITLE
Reworking code for Java 17

### PR DESCRIPTION
**Pull request by novitpw**
Beginning with Java versions 14 and 16, the "switch" command and the "instanceof" operator have been redesigned. Also in Java 16 a tuple called "record" was added. More information can be found at [JDK Enhancement Proposal](https://openjdk.org/projects/jdk/17/jeps-since-jdk-11) (JEP 361, 394, 395).

### CODE REVIEW

File: native/src/main/java/com/velocitypowered/natives/util/MoreByteBufUtils.java [hunk 1]
```diff
@@ -77,15 +72,9 @@ private static boolean isCompatible(Native nativeStuff, ByteBuf buf) {
...
-        return alloc.heapBuffer(initialCapacity);
-      case DIRECT_PREFERRED:
-      case DIRECT_REQUIRED:
-        return alloc.directBuffer(initialCapacity);
-      default:
-        throw new AssertionError("Preferred buffer type unknown");
-    }
+    return switch (nativeStuff.preferredBufferType()) {
+      case HEAP_REQUIRED, HEAP_PREFERRED -> alloc.heapBuffer(initialCapacity);
+      case DIRECT_PREFERRED, DIRECT_REQUIRED -> alloc.directBuffer(initialCapacity);
```
Comment (Loyisa) [inline native/src/main/java/com/velocitypowered/natives/util/MoreByteBufUtils.java:77]:
```java
default -> throw new AssertionError("Preferred buffer type unknown");
```

**Review by 4drian3d — APPROVED** [review]

**Review by Loyisa — CHANGES_REQUESTED** [review]

File: native/src/main/java/com/velocitypowered/natives/util/MoreByteBufUtils.java (hunk 1 above)
Comment (novitpw) [reply native/src/main/java/com/velocitypowered/natives/util/MoreByteBufUtils.java:77]: Will never be used. In cases where enum is updated, an error immediately appears stating that the "switch" statement does not cover all possible input values.
image

File: proxy/src/main/java/com/velocitypowered/proxy/protocol/netty/GameSpyQueryHandler.java [hunk 2]
```diff
@@ -212,12 +210,9 @@ private List<QueryResponse.PluginInformation> getRealPluginInformation() {
...
   }
 
-  private static class ResponseWriter {
-
-    private final ByteBuf buf;
-    private final boolean isBasic;
+  private record ResponseWriter(ByteBuf buf, boolean isBasic) {
 
-    ResponseWriter(ByteBuf buf, boolean isBasic) {
+    private ResponseWriter(ByteBuf buf, boolean isBasic) {
```
Comment (4drian3d) [inline proxy/src/main/java/com/velocitypowered/proxy/protocol/netty/GameSpyQueryHandler.java]: To do validations on a constructor of a record, you can simply do this

image

**Review by 4drian3d — APPROVED** [review]

**Comment by ham1255** [comment]
it should fix #1080 i think

**Review by RealBauHD — APPROVED** [review]

File: api/src/main/java/com/velocitypowered/api/command/CommandManager.java [hunk 3]
```diff
@@ -43,7 +43,7 @@ public interface CommandManager {
    * @param otherAliases additional aliases
    * @throws IllegalArgumentException if one of the given aliases is already registered, or
    *         the given command does not implement a registrable {@link Command} subinterface
-   * @see Command for a list of registrable {@link Command} subinterfaces
+   * @see Command for a list of registrable Command subinterfaces
```
Comment (RealBauHD) [inline api/src/main/java/com/velocitypowered/api/command/CommandManager.java:46]: I don't see any reason for this

File: api/src/main/java/com/velocitypowered/api/command/CommandManager.java [hunk 4]
```diff
@@ -64,7 +64,7 @@ default void register(String alias, Command command, String... otherAliases) {
    * @param command the command to register
    * @throws IllegalArgumentException if one of the given aliases is already registered, or
    *         the given command does not implement a registrable {@link Command} subinterface
-   * @see Command for a list of registrable {@link Command} subinterfaces
+   * @see Command for a list of registrable Command subinterfaces
```
Comment (RealBauHD) [inline api/src/main/java/com/velocitypowered/api/command/CommandManager.java:67]: same here

**Comment by novitpw** [comment]
It will never be merged...

**Pull request closed without merging**